### ISSUE TITLE
Flavors: add `gh-aw` flavor support for GitHub Agentic Workflows

### ISSUE THREAD

**Issue by pygarap** [rvben/rumdl#405]
## Summary

Add a new Markdown flavor, `gh-aw`, tuned for GitHub Agentic Workflows source files (`.github/workflows/*.md`) that combine YAML frontmatter with markdown instructions.

## Motivation

`gh-aw` workflow sources are markdown files that start with YAML frontmatter wrapped in `---`, then a normal markdown body. They also support workflow-specific templating directives (conditionals, compile-time imports, runtime imports). Without a dedicated flavor, rumdl can produce false positives on valid `gh-aw` constructs.

### Flavor name
- New flavor: `gh-aw`

### Syntax support to treat as valid
1. YAML frontmatter at the top of the file:
   - Treat the leading `--- ... ---` block as frontmatter (not a thematic break).
   - Allow frontmatter before the first heading without triggering "document must start with a heading" style rules.

2. `gh-aw` templating directives in markdown body:
   - Conditional blocks:
     - `{{#if ...}}` and `{{/if}}`
   - Runtime import macros:
     - `{{#runtime-import path[:start-end]}}`
     - `{{#runtime-import? path[:start-end]}}`
   - Compile-time imports (if present in markdown body):
     - `{{#import ...}}`

### Configuration and docs
- Document the flavor in `docs/flavors.md` and add `docs/flavors/gaw.md` describing:
  - Intended file locations (`.github/workflows/*.md`, optionally `.github/agents/*.md` if applicable).
  - Which rules are adjusted and why.
- No new auto-detection is required (these files are still `*.md`), but an explicit mention of the recommended `[per-file-flavor]` pattern for `.github/workflows/**/*.md` would help.

## Acceptance criteria

- A minimal `gh-aw` workflow example (frontmatter + headings + `{{#if}}` + `{{#runtime-import}}`) passes `rumdl check` under `flavor = "gaw"` with no flavor-related false positives.
- Formatter output remains stable (no rewrites that would break the directive lines or frontmatter fences).
- Docs clearly list rule deltas for `gaw`.

## References

- https://github.com/github/gh-aw/blob/main/README.md
- https://github.com/github/gh-aw/blob/main/AGENTS.md
- https://github.github.com/gh-aw/reference/workflow-structure/
- https://github.github.com/gh-aw/reference/templating/
- https://github.github.com/gh-aw/reference/imports/
- https://github.github.com/gh-aw/reference/markdown/

**Comment by rvben** [comment]
Thanks for the detailed request, @pygarap. GitHub Agentic Workflows looks promising, but since it's still in technical preview and the spec may change significantly, we're deferring dedicated flavor support until it stabilizes.

In the meantime, you can handle the false positives with `per-file-ignores`:
```toml
[per-file-ignores]
".github/workflows/*.md" = ["MD041", "MD057"]
```

We'll revisit when gh-aw reaches GA.

**Comment by rvben** [comment]
Implemented and released in rumdl v0.2.64 with preview support for the `gh-aw` flavor.

Announcement and demo: https://x.com/rvbenam8/status/2095483319282532725?s=20

Thanks again for the original request, @pygarap!
